dace/libraries/pblas/environments/intel_mkl_mpich.py

```diff
@@ -0,0 +1,156 @@
+
+        if 'CONDA_PREFIX' in os.environ:
+            warnings.warn('Anaconda Python is installed but the MKL library file cannot be found for linkage. Please '
+                          'install libraries with "conda install mkl-devel" or set the MKLROOT environment variable.')
+        else:
+            warnings.warn('MKL was not found. Please install MKL or set hte MKLROOT environment variable.')
+
+        return ""
+
+    @staticmethod
+    def cmake_includes():
+        if 'MKLROOT' in os.environ:
+            return [os.path.join(os.environ['MKLROOT'], 'include')]
+        # Anaconda
+        elif 'CONDA_PREFIX' in os.environ:
+            base_path = os.environ['CONDA_PREFIX']
+            # Anaconda on Windows
+            candpath = os.path.join(base_path, 'Library', 'include')
+            if os.path.isfile(os.path.join(candpath, 'mkl.h')):
+                return [candpath]
+            # Anaconda on other platforms
+            candpath = os.path.join(base_path, 'include')
+            if os.path.isfile(os.path.join(candpath, 'mkl.h')):
+                return [candpath]
+            warnings.warn('Anaconda Python is installed but the MKL include directory cannot '
+                          'be found. Please install MKL includes with '
+                          '"conda install mkl-include" or set the MKLROOT environment '
+                          'variable')
+            return []
+        else:
+            return []
+
+    @staticmethod
+    def cmake_libraries():
+
+        libpath = IntelMKLScaLAPACKMPICH._find_mkl_lib_path()
+
+        prefix = Config.get('compiler', 'library_prefix')
+        suffix = Config.get('compiler', 'library_extension')
+
+        libfiles = [os.path.join(libpath, f"{prefix}{name}.{suffix}") for name in IntelMKLScaLAPACKMPICH.libraries]
+
+        simd_libfile = os.path.join(libpath, f"{prefix}{IntelMKLScaLAPACKMPICH.simd}.{suffix}")
+        if not os.path.isfile(simd_libfile):
+            for num in range(1, 6):
+                simd_libfile = os.path.join(libpath, f"{prefix}{IntelMKLScaLAPACKMPICH.simd}.{suffix}.{num}")
+                if os.path.isfile(simd_libfile):
+                    break
+        if os.path.isfile(simd_libfile):
+            libfiles.append(simd_libfile)
+
+        return libfiles
+
+    @staticmethod
+    def cmake_link_flags():
+
+        # From https://stackoverflow.com/questions/15008758/parsing-boolean-values-with-argparse
+        def str2bool(v: Union[None, str, bool]) -> bool:
+            if v is None:
+                return False
+            if isinstance(v, bool):
+                return v
+            if isinstance(v, str):
+                if v.lower() in ('yes', 'true', 't', 'y', '1'):
+                    return True
+                if v.lower() in ('no', 'false', 'f', 'n', '0'):
+                    return False
+            return False
+
+        is_daint = str2bool(os.getenv('IS_DAINT'))
```

Comment:
oh no

Reply:
We are keeping this as an example for adjusting the environment to work on different systems. It is impossible to work around this with the current environment API in an elegant way.